common/persistence/historyV2Store.go

```diff
@@ -79,6 +79,7 @@ func (m *historyV2ManagerImpl) ForkHistoryBranch(request *ForkHistoryBranchReque
 		ForkNodeID:     request.ForkNodeID,
 		NewBranchID:    uuid.New(),
 		Info:           request.Info,
+		ShardID:        request.ShardID,
```

Comment:
I am a bit worried about missing some places in application layer. Can you add an assertion here to require shardID? return error if that is empty.

Reply:
This store is used by Cassandra and Mysql. How about adding the assertion in sqlHistoryV2Manager.go?

Reply:
I think it should be in the Store, even though Cassandra doesn't need it, it is still required as part of the contract. What we want is to keep our persistence interface as simple as possible, and centralize any logic that can be shared by all different DBs.

Reply:
Make sense. Added the validation logic.